cluster/gce/upgrade.sh

```diff
@@ -362,95 +458,30 @@ function do-node-upgrade() {
-      fi
 
-      # Recreate instance
-      echo "== Recreating instance ${instance}. ==" >&2
-      recreate=$(gcloud compute instance-groups managed recreate-instances "${group}" \
-        --project="${PROJECT}" \
-        --zone="${ZONE}" \
-        --instances="${instance}" 2>&1) && recreate_rc=$? || recreate_rc=$?
-      if [[ "${recreate_rc}" != 0 ]]; then
-        echo "== FAILED to recreate ${instance} =="
-        echo "${recreate}"
-        return ${recreate_rc}
-      fi
-
-      # Wait for instance to be recreated
-      echo "== Waiting for instance ${instance} to be recreated. ==" >&2
-      while true; do
-        new_instance_id=$(gcloud compute instances describe "${instance}" \
-          --format='get(id)' \
-          --project="${PROJECT}" \
-          --zone="${ZONE}" 2>&1) && describe_rc=$? || describe_rc=$?
-        if [[ "${describe_rc}" != 0 ]]; then
-          echo "== FAILED to describe ${instance} =="
-          echo "${new_instance_id}"
-          echo "  (Will retry.)"
-        elif [[ "${new_instance_id}" == "${instance_id}" ]]; then
-          echo -n .
-        else
-          echo "Instance ${instance} recreated."
-          break
-        fi
-        sleep 1
-      done
-
-      # Wait for k8s node object to reflect new instance id
-      echo "== Waiting for new node to be added to k8s.  ==" >&2
-      while true; do
-        external_id=$("${KUBE_ROOT}/cluster/kubectl.sh" get node "${instance}" --output=jsonpath='{.spec.externalID}' 2>&1) && kubectl_rc=$? || kubectl_rc=$?
-        if [[ "${kubectl_rc}" != 0 ]]; then
-          echo "== FAILED to get node ${instance} =="
-          echo "${external_id}"
-          echo "  (Will retry.)"
-        elif [[ "${external_id}" == "${new_instance_id}" ]]; then
-          echo "Node ${instance} recreated."
-          break
-        elif [[ "${external_id}" == "${instance_id}" ]]; then
-          echo -n .
-        else
-          echo "Unexpected external_id '${external_id}' matches neither old ('${instance_id}') nor new ('${new_instance_id}')."
-          echo "  (Will retry.)"
-        fi
-        sleep 1
-      done
-
-      # Wait for the node to not have SchedulingDisabled=True and also to have
-      # Ready=True.
-      echo "== Waiting for ${instance} to become ready. ==" >&2
-      while true; do
-        cordoned=$("${KUBE_ROOT}/cluster/kubectl.sh" get node "${instance}" --output='jsonpath={.status.conditions[?(@.type == "SchedulingDisabled")].status}')
-        ready=$("${KUBE_ROOT}/cluster/kubectl.sh" get node "${instance}" --output='jsonpath={.status.conditions[?(@.type == "Ready")].status}')
-        if [[ "${cordoned}" == 'True' ]]; then
-          echo "Node ${instance} is still not ready: SchedulingDisabled=${ready}"
-        elif [[ "${ready}" != 'True' ]]; then
-          echo "Node ${instance} is still not ready: Ready=${ready}"
-        else
-          echo "Node ${instance} Ready=${ready}"
-          break
+    proc=${node_upgrade_parallelism}
```

Comment:
Just a counter for no. of more processes that can be started in parallel. Renamed it to `process_count_left`.